### PR TITLE
Add spell checker to CI

### LINKED ISSUE #147: add typo checker to CI
Typos are annoying. Let's add a spell checker to our CI.

https://crates.io/crates/typos looks cool, maybe that one? I like how it's a bit more tailored than a full-on spell check.

### PR DESCRIPTION
**Pull request by Aadamandersson**
Added a spell checker (https://crates.io/crates/typos) to our CI.
Closes #147.

### CODE REVIEW

File: .typos.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[files]
+extend-exclude = ["components/dada-web/ace/", "book/src/css/speech-bubbles.css", "book/package-lock.json"]
+[default.extend-words]
+fo = "fo"
```
Comment (XFFXFF) [inline .typos.toml:4]: What is this for?

**Review by XFFXFF — COMMENTED** [review]
Nice!  
left a question

File: .typos.toml (hunk 1 above)
Comment (Aadamandersson) [reply .typos.toml:4]: `extend-exclude` excludes the listed files from consideration and will therefore not be checked for spelling mistakes. I.e., allow typos in third party files + package-lock.json.
`extend-words` allows the listed words and will not treat them as a typo. This was added to allow the examples here:
https://github.com/dada-lang/dada/blob/main/book/docs/reference/raw-string-literals.md

**Review by XFFXFF — APPROVED** [review]
Thanks!  
bors r+

**Pull request merged**